viewing_party/party.py

```diff
@@ -1,23 +1,276 @@
+    action_count = 0
+    intrigue_count = 0
+    
+    if len(user_data["watched"]) < 1:
+        most_watched_genre = None 
+        return most_watched_genre
+
+    for movie in user_data["watched"]:
+        if movie["genre"] == "Fantasy":
+            fantasy_count += 1
+        if movie["genre"] == "Action":
+            action_count += 1
+        if movie["genre"] == "Intrigue":
+            intrigue_count += 1
+
+    if fantasy_count > action_count and fantasy_count > intrigue_count:
+        most_watched_genre = "Fantasy"
+    if action_count > fantasy_count and action_count > intrigue_count:
+        most_watched_genre = "Action"
+    if intrigue_count > action_count and intrigue_count > fantasy_count:
+        most_watched_genre = "Intrigue"
+    return most_watched_genre
+
+
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):
+    user_watched_list = []
+    friend_watched_list = []
+
+    for movie in user_data["watched"]:
+        user_watched_list.append(movie["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_watched_list.append(movie["title"])
+    
+    user_watched_set = set(user_watched_list)
+    friend_watched_set = set(friend_watched_list)
+    user_unique_watched_set = user_watched_set.difference(friend_watched_set)
+    user_unique_watched_list = []
+
+    for unique_movie in user_unique_watched_set:
+        for movie in user_data["watched"]:
+            if movie["title"] == unique_movie:
+                user_unique_watched_list.append(movie)
+
+    return user_unique_watched_list
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = []
+    friend_watched_list = []
+
+    for movie in user_data["watched"]:
+        user_watched_list.append(movie["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_watched_list.append(movie["title"])
+    
+    user_watched_set = set(user_watched_list)
+    friend_watched_set = set(friend_watched_list)
+    friend_unique_watched_set = friend_watched_set.difference(user_watched_set)
+    friend_unique_watched_list = []
+
+    for unique_movie in friend_unique_watched_set:
+        for friend in user_data["friends"]:
+            for movie in friend["watched"]:
+                if movie not in friend_unique_watched_list:
+                    if movie["title"] == unique_movie:
+                        friend_unique_watched_list.append(movie)
+
+    return friend_unique_watched_list
+
+
+##I know if i reverse this somehow it would work more eloquently but i couldnt seem to make that work
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    friend_watched_movie_title_list = []

```

Comment:
It looks like a lot of the logic here is similar to the get unique methods from wave 3. Can we reuse those functions to help simplify the implementation here? Part of the description says that the recommendations should be things the user _hasn't_ watched, but that at least one friend _has_ watched. Does that sound like the results of one of the wave 3 functions?